engine/access/rpc/backend/backend_network.go

```diff
@@ -80,6 +83,71 @@ func (b *backendNetwork) GetLatestProtocolStateSnapshot(_ context.Context) ([]by
 	return data, nil
 }
 
+// GetProtocolStateSnapshotByBlockID returns serializable Snapshot by blockID
+func (b *backendNetwork) GetProtocolStateSnapshotByBlockID(_ context.Context, blockID flow.Identifier) ([]byte, error) {
+	snapshotByBlockId := b.state.AtBlockID(blockID)
+	snapshotHeadByBlockId, err := snapshotByBlockId.Head()
```

Comment:
nit: the snapshot isn't needed
```suggestion
	snapshotHeadByBlockId, err := b.state.AtBlockID(blockID).Head()
```